assembl/tasks/common.py

```diff
@@ -0,0 +1,321 @@
+import sys
+import os
+import os.path
+from contextlib import nested
+import base64
+import json
+
+from invoke import Task, task as base_task
+from os.path import dirname, realpath
+DEFAULT_SECTION = "DEFAULT"
+_local_file = __file__
+if _local_file.endswith('.pyc'):
+    _local_file = _local_file[:-1]
+local_code_root = dirname(dirname(realpath(_local_file)))
+
+
+def task(*args, **kwargs):
+    pre = list(kwargs.pop('pre', args))
+    pre.append(base_task(setup_ctx))
+    return base_task(pre=pre, **kwargs)
+
+
+def exists(c, path, sudo=False):
+    """
+    Return True if given path exists on the current remote host. Taken from patchwork.
+    """
+    cmd = '{}test -e "$(echo {})"'.format('sudo ' if sudo else '', path)
```

Comment:
Why not use built-in python utils to check if file exists (such as os.path.exists) ?

Reply:
There were a few reasons. But I think the only one of the reasons that it remains is the sudo aspect, allowing `sudo` checking if a file exists. `os.path.exists` does not have those kinds of rights.